src/Kunstmaan/NodeBundle/Repository/NodeRepository.php

```diff
@@ -327,12 +345,25 @@ public function getAllMenuNodes(
         $qb = $aclNativeHelper->apply($qb, $permissionDef);
 
         $stmt = $this->_em->getConnection()->prepare($qb->getSQL());
+
+        if (!$includeHiddenFromNav) {
+            $stmt->bindValue(':hiddenFromNavFalse', false);
+        }
         $stmt->bindValue(':lang', $lang);
         if (!\is_null($rootNode)) {
             $stmt->bindValue(':left', $rootNode->getLeft());
             $stmt->bindValue(':right', $rootNode->getRight());
         }
         $stmt->execute();
+        if ($databasePlatformName == 'postgresql') {
```

Comment:
This is something I would like to solve in the query, as it could potentially a huge array to be looped over on large websites. If we can "pre-filter" these results in the query, this heavy lifting is done by the database